theatre/studio/src/panels/SequenceEditorPanel/DopeSheet/Right/BasicKeyframedTrack/BasicKeyframedTrack.tsx

```diff
@@ -41,11 +67,43 @@ const BasicKeyframedTrack: React.FC<{
       layoutP={layoutP}
       leaf={leaf}
       key={'keyframe-' + kf.id}
-      selection={selectedKeyframeIds[kf.id] === true ? selection : undefined}
+      selection={selectedKeyframeIds?.[kf.id] === true ? selection : undefined}
     />
   ))
 
-  return <>{keyframeEditors}</>
+  return (
+    <TrackContainer
+      ref={ref}
+      highlight={Boolean(trackId && trackToHighlightId === trackId)}
+    >
+      {contextMenu}
+      {keyframeEditors}
+    </TrackContainer>
+  )
 })
 
+function useTrackContextMenu(
+  node: HTMLDivElement | null,
+  {
+    leaf,
+  }: {
+    leaf: SequenceEditorTree_PrimitiveProp
+  },
+) {
+  const copiedKeyframes = getCopiedKeyframes()
```

Comment:
I think we can put all of this inside the callback of `useContextMenu()`. 
<img width="274" alt="grafik" src="https://user-images.githubusercontent.com/593118/154458432-3c3fb964-bb43-44a4-bf58-55dfa90698cf.png">

Basically, we don't need to calculate this stuff unless the user has triggered the context menu.

Reply:
Yes that would make a lot more sense